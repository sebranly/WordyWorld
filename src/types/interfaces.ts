export interface Word {
  en: string;
  fr: string;
}
