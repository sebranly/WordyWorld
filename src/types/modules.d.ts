declare module "emoji-from-text" {
  function emojiFromText(text: string, onlyOneMatch?: boolean): any;
  export = emojiFromText;
}
