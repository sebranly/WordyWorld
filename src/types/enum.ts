enum Screen {
  About,
  Explore,
  Game
}

enum WordConnection {
  Addition,
  Anagram,
  Deletion,
  Replacement,
  Same
}

export { Screen, WordConnection };
