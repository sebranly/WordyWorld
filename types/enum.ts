enum Screen {
  About,
  Config,
  Explore
}

export { Screen };
