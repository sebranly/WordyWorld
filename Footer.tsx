import * as React from "react";
import {
  FooterTab,
  Footer as FooterNativeBase,
  Icon,
  Text,
  Button
} from "native-base";

import { Screen } from "./types/enum";

export interface FooterProps {
  screen: Screen;
}

const Footer: React.FC<FooterProps> = props => {
  // Setup
  const { screen } = props;
  const isAbout = screen === Screen.About;
  const isConfig = screen === Screen.Config;
  const isExplore = screen === Screen.Explore;

  // TODO: add `onPress` on `Button`
  return (
    <FooterNativeBase>
      <FooterTab>
        <Button vertical active={isConfig}>
          <Icon name="ios-settings" />
          <Text>Config</Text>
        </Button>
        <Button vertical active={isExplore}>
          <Icon name="md-globe" />
          <Text>Explore</Text>
        </Button>
        <Button vertical active={isAbout}>
          <Icon name="ios-information-circle" />
          <Text>About</Text>
        </Button>
      </FooterTab>
    </FooterNativeBase>
  );
};

export { Footer };
