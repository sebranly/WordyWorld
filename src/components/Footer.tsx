// Vendor
import * as React from "react";
import {
  FooterTab,
  Footer as FooterNativeBase,
  Icon,
  Text,
  Button
} from "native-base";

// Internal
import { Screen } from "../types/enum";

export interface FooterProps {
  changeScreen: (screen: Screen) => void;
  screen: Screen;
}

const Footer: React.FC<FooterProps> = (props) => {
  // Setup
  const { changeScreen, screen } = props;

  const renderButton = (buttonScreen: Screen) => {
    const isActive = buttonScreen === screen;

    let iconName = "";
    let text = "";

    switch (buttonScreen) {
      case Screen.About:
        iconName = "ios-information-circle";
        text = "About";
        break;

      case Screen.Game:
        iconName = "ios-play-circle";
        text = "Play";
        break;

      case Screen.Explore:
      default:
        iconName = "md-globe";
        text = "Explore";
        break;
    }

    return (
      <Button
        key={buttonScreen}
        vertical
        active={isActive}
        onPress={() => changeScreen(buttonScreen)}
      >
        <Icon name={iconName} />
        <Text>{text}</Text>
      </Button>
    );
  };

  const buttons = [Screen.Explore, Screen.Game, Screen.About].map((v) =>
    renderButton(v)
  );

  return (
    <FooterNativeBase>
      <FooterTab>{buttons}</FooterTab>
    </FooterNativeBase>
  );
};

export { Footer };
