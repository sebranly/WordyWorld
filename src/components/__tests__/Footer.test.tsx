// Vendor
import React from "react";
import renderer from "react-test-renderer";

// Internal
import { Footer } from "../Footer";
import { Screen } from "../../types/enum";

it("renders", () => {
  const props = { changeScreen: jest.fn(), screen: Screen.About };
  const rendered = renderer.create(<Footer {...props} />).toJSON();
  expect(rendered).not.toBeNull();
  expect(rendered).toMatchSnapshot();
});
