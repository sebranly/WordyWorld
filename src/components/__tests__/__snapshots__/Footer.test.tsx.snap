// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "paddingBottom": 0,
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": undefined,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <
        active={false}
        activeOpacity={0.5}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        vertical={true}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#6b6b6b",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Explore
        </Text>
      </>
      <
        active={false}
        activeOpacity={0.5}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        vertical={true}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#6b6b6b",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Play
        </Text>
      </>
      <
        active={true}
        activeOpacity={0.5}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#cde1f9",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        vertical={true}
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 24,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingTop": 2,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          About
        </Text>
      </>
    </View>
  </View>,
]
`;
